src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/TemplateBuilder.cs

```diff
@@ -125,10 +125,17 @@ public IHtmlContent Build()
             var formatString = _readOnly ?
                 viewData.ModelMetadata.DisplayFormatString :
                 viewData.ModelMetadata.EditFormatString;
+
             if (_model != null && !string.IsNullOrEmpty(formatString))
             {
                 formattedModelValue = string.Format(CultureInfo.CurrentCulture, formatString, formattedModelValue);
             }
+            else if (string.Equals("week", _templateName, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
This block should be moved up and connected to the conditions ending at https://github.com/aspnet/Mvc/pull/7045/files#diff-ba280b63c56b14f4e63ae0bc9f11f7eaR123

The idea of the https://github.com/aspnet/Mvc/pull/7045/files#diff-ba280b63c56b14f4e63ae0bc9f11f7eaR132
block is to override whatever we come up with when `DisplayFormatString` or `EditFormatString` is set.

Reply:
And, yes, I know the current code is a bit odd. Think the original idea was to get other information as well as `formattedModelValue` in the previous block.